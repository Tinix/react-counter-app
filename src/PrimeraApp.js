import React from 'react';
import PropTypes from 'prop-types';

const PrimeraApp = ({ saludo, subtitle }) => {
  


  return (
    <>
      <h1> { saludo } </h1>
      {/*<pre> { JSON.stringify(saludo, null 3) }</pre>*/}
      <p>{ subtitle }</p>
    </>
    );
}

PrimeraApp.protoTypes = {
  saludo: PropTypes.string.isRequired
 }


PrimeraApp.defaultProps = {
  subtitle: 'Soy un subtitulo'
}


export default PrimeraApp;
