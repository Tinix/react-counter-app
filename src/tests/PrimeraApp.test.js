import React from 'react';
import PrimeraApp from '../PrimeraApp';
import '@testing-library/jest-dom';
import {shallow} from 'enzyme';



describe("Prueba en <PrimeraAPP />", () => {
 // test('debe demostrar el mensaje "Hola Soy Goku" ',() => {
    //const saludo = "Hola soy Goku";
    //const { getByText }  = render( <PrimeraApp saludo={ saludo } />);
    //expect(getByText(saludo)).toBeInTheDocument();
 // });
  
  test("debe de mostrar  <PrimerApp> correctamente", () =>{

   const saludo = 'Hola , Soy Goku';

    const wrapper = shallow(<PrimeraApp  saludo={ saludo } />)

    expect(wrapper).toMatchSnapshot();
 });

  test("debe de mostrar el sustitulo enviado por props", () => {
    
    const saludo = "Hola, Soy Goku";
    const subTitulo = "Soy un subtitulo";

    const wrapper = shallow(
      <PrimeraApp
	saludo={ saludo }
	subtitulo={ subTitulo }
      />
    );

    const textoParrafo = wrapper.find('p').text();
    // console.log(textoParrafo);
    expect(textoParrafo).toBe(subTitulo);
   });

});
