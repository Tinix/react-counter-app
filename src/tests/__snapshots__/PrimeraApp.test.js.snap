// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueba en <PrimeraAPP /> debe de mostrar  <PrimerApp> correctamente 1`] = `
<Fragment>
  <h1>
     
    Hola , Soy Goku
     
  </h1>
  <p>
    Soy un subtitulo
  </p>
</Fragment>
`;
